gunicorn/arbiter.py

```diff
@@ -186,10 +187,11 @@ def init_signals(self):
         # initialize all signals
         for s in self.SIGNALS:
             signal.signal(s, self.signal)
-        signal.signal(signal.SIGCHLD, self.handle_chld)
```

Comment:
This part is a mistake. We still need to install this handler separately, and it should not be in the list of `SIGNALS` above, if we are not going to use the queue for it.

Reply:
Sorry, that's wrong. The point of this patch is to handle reaping in the main thread.

What's missing is a check for `self.SIGCHLD` in the `run` method, with a call `self.handle_chld` from the main loop.